src/assets/CurrencyRateController.ts

```diff
@@ -195,16 +195,27 @@ export class CurrencyRateController extends BaseController<
       pendingCurrentCurrency,
       pendingNativeCurrency,
     } = this.state;
+
+    const conversionDate: number = Date.now() / 1000;
+    let conversionRate: number | null = null;
+    let usdConversionRate: number | null = null;
     try {
-      const {
-        conversionDate,
-        conversionRate,
-        usdConversionRate,
-      } = await this.fetchExchangeRate(
-        pendingCurrentCurrency || currentCurrency,
-        pendingNativeCurrency || nativeCurrency,
-        this.includeUsdRate,
-      );
+      if (
+        (pendingCurrentCurrency || currentCurrency) &&
+        (pendingNativeCurrency || nativeCurrency) &&
+        pendingNativeCurrency !== ''
+      ) {
+        ({ conversionRate, usdConversionRate } = await this.fetchExchangeRate(
+          pendingCurrentCurrency || currentCurrency,
+          pendingNativeCurrency || nativeCurrency,
+          this.includeUsdRate,
+        ));
+      }
+    } catch (error) {
+      if (!error.message.includes('market does not exist for this coin pair')) {
+        throw error;
+      }
+    } finally {
       this.update(() => {
```

Comment:
I wonder if `this.update` could fail :thinking: 

It might be worth wrapping this in _another_ `try... finally` just to be extra sure that `releaseLock` is called. I don't know if this can fail or not, but best to stay on the safe side when the consequence is a deadlock.